test/LiveDeploy.t.sol

```diff
@@ -0,0 +1,359 @@
+        if (block.chainid == 31_337) {
+            forkId = vm.createFork(vm.envString(rpcKey));
+            vm.selectFork(forkId);
+        }
+    }
+}
+
+contract LiveDeploy is ForkTest, DeployAll {
+    using FixedPointMathLib for uint256;
+
+    ERC20 constant NATIVE_ERC20 = ERC20(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    uint256 ONE_SHARE;
+    uint8 constant SOLVER_ROLE = 42;
+
+    function setUp() public virtual {
+        string memory FILE_NAME;
+        if (block.chainid == 31_337) {
+            FILE_NAME = "exampleL1.json";
+        } else {
+            FILE_NAME = vm.envString("LIVE_DEPLOY_READ_FILE_NAME");
+        }
+
+        // we have to start the fork again... I don't exactly know why. But it's a known issue with foundry re:
+        // https://github.com/foundry-rs/foundry/issues/5471
+        _startFork(DEFAULT_RPC_URL);
+        // (new DeployRateProviders()).run("liveDeploy", FILE_NAME, true);
+        // Run the deployment scripts
+
+        run(FILE_NAME);
+        // warp forward the minimumUpdateDelay for the accountant to prevent it from pausing on update test
+        vm.warp(block.timestamp + mainConfig.minimumUpdateDelayInSeconds);
+
+        // define one share based off of vault decimals
+        ONE_SHARE = 10 ** BoringVault(payable(mainConfig.boringVault)).decimals();
+
+        // give this the SOLVER_ROLE to call bulkWithdraw
+        RolesAuthority rolesAuthority = RolesAuthority(mainConfig.rolesAuthority);
+        vm.startPrank(mainConfig.protocolAdmin);
+        rolesAuthority.setUserRole(address(this), SOLVER_ROLE, true);
+        rolesAuthority.setRoleCapability(
+            SOLVER_ROLE, mainConfig.teller, TellerWithMultiAssetSupport.bulkWithdraw.selector, true
+        );
+        vm.stopPrank();
+    }
+
+    function testDepositAndBridge(uint256 amount) public {
+        string memory tellerName = mainConfig.tellerContractName;
+        if (compareStrings(tellerName, "CrossChainOPTellerWithMultiAssetSupport")) {
+            _testOPDepositAndBridge(ERC20(mainConfig.base), amount);
+        } else if (compareStrings(tellerName, "MultiChainLayerZeroTellerWithMultiAssetSupport")) {
+            _testLZDepositAndBridge(ERC20(mainConfig.base), amount);
+        } else { }
+    }
+
+    function testDepositBaseAssetAndUpdateRate(uint256 depositAmount, uint96 rateChange) public {
+        AccountantWithRateProviders accountant = AccountantWithRateProviders(mainConfig.accountant);
+        // manual bounding done because bound() doesn't exist for uint96
+        rateChange = rateChange % uint96(mainConfig.allowedExchangeRateChangeUpper - 1);
+        rateChange = (rateChange < mainConfig.allowedExchangeRateChangeLower + 1)
+            ? mainConfig.allowedExchangeRateChangeLower + 1
+            : rateChange;
+
+        // mint a bunch of extra tokens to the vault for if rate increased
+        deal(mainConfig.base, mainConfig.boringVault, depositAmount);
```

Comment:
What happens if the fuzzed `depositAmount` at this step is 0? And then the `depositAmount` gets bounded to non-zero, so could there not be extra assets in the vault at the final step to be able to withdraw an increased amount of assets?